pkg/vault/jxvaulter.go

```diff
@@ -0,0 +1,45 @@
+package vault
+
+import (
+	"fmt"
+	"github.com/hashicorp/vault/api"
+	"net/url"
+)
+
+type JxVaulter struct {
+	client *api.Client
+}
+
+func newJxVaulter(client *api.Client) Vaulter {
+	return &JxVaulter{client: client}
+}
+
+func NewVaulter(o VaultOptions) (Vaulter, error) {
+	clientFactory, err := NewVaultClientFactory(o)
+	client, err := clientFactory.NewVaultClient(o.VaultName(), o.VaultNamespace())
+	return newJxVaulter(client), err
+}
+
+func (v *JxVaulter) Config() (vaultUrl url.URL, vaultToken string, err error) {
+	parsed, err := url.Parse(v.client.Address())
+	return *parsed, v.client.Token(), err
+}
+
+func (v *JxVaulter) Secrets() ([]string, error) {
```

Comment:
Wasn't be better to return a `map[string]string` where the key of the map is the secret name and the value is the value of the secret. We will definitely need this.

Reply:
The underlying vault store can support more than just a simple string-->string mapping.
ie, any given secret can have any format supported by JSON. That's why I left it as just the list of secret names for the moment. We can expand/fix this when we start storing secrets in vault. I didn't want to guess ahead of time how we may store secrets without knowing what secrets we need to storre

Reply:
Is it the returned type a string, even though the content type is some serialised JSON. Or is it an interface{}?

Reply:
The returned type is a `string` (being the name of the secret). This method doesn't return the underlying data.